src/foundry/common/documents/token.d.mts

```diff
@@ -43,55 +43,141 @@ declare abstract class BaseToken extends Document<"Token", BaseToken.Schema, any
    *     delete: "TOKEN_DELETE"
    *   },
-   *   schemaVersion: "12.324"
+   *   schemaVersion: "13.341"
    * })
    * ```
    */
   static override metadata: BaseToken.Metadata;
 
   static override defineSchema(): BaseToken.Schema;
 
+  /** @defaultValue `["DOCUMENT", "TOKEN"]` */
+  static override LOCALIZATION_PREFIXES: string[];
+
+  /** @defaultValue `["x", "y", "elevation", "width", "height", "shape"]` */
+  static MOVEMENT_FIELDS: Readonly<string[]>;
+
   /**
    * The default icon used for newly created Token documents
    * @defaultValue `CONST.DEFAULT_TOKEN`
    */
   static DEFAULT_ICON: string;
 
   /**
-   * @remarks If `this.actor`, uses {@link Actor.testUserPermission | `this.actor.testUserPermission`} otherwise `super`'s. Core's `Actor` implementation
-   * doesn't override this method, so without further extension, that's equivalent to {@link Document.testUserPermission | `Document#testUserPermission`}
+   * Prepare changes to a descendent delta collection.
+   * @param changes - Candidate source changes.
+   * @param options - Options which determine how the new data is merged.
    */
-  // options: not null (destructured)
-  override testUserPermission(
-    user: User.Implementation,
-    permission: Document.ActionPermission,
-    options?: Document.TestUserPermissionOptions,
-  ): boolean;
+  protected _prepareDeltaUpdate(changes?: TokenDocument.UpdateData, options?: DataModel.UpdateOptions): void;
 
   // changes, options: not null (parameter default only)
   override updateSource(
     changes?: TokenDocument.UpdateData,
     options?: DataModel.UpdateOptions,
   ): TokenDocument.UpdateData;
 
+  /**
+   * Get the snapped position of the Token.
+   * @param data - The position and dimensions
+   * @returns The snapped position
+   */
+  getSnappedPosition(
+    data?: InexactPartial<foundry.canvas.Canvas.ElevatedPoint & TokenDocument.Dimensions>,
+  ): foundry.canvas.Canvas.ElevatedPoint;
+
+  /**
+   * Get the top-left offset of the Token
+   * @param data - The position and dimensions
+   * @returns The top-left grid offset
+   */
+  protected _positionToGridOffset(
+    data?: InexactPartial<foundry.canvas.Canvas.ElevatedPoint & TokenDocument.Dimensions>,
+  ): foundry.grid.BaseGrid.Offset3D;
+
+  /**
+   * Get the position of the Token from the top-left grid offset.
+   * @param offset - The top-left grid offset
+   * @param data   - The dimensions that override the current dimensions
+   * @returns The snapped position
+   */
+  protected _gridOffsetToPosition(
+    offset: foundry.grid.BaseGrid.Offset3D,
+    data?: InexactPartial<TokenDocument.Dimensions>,
+  ): foundry.canvas.Canvas.ElevatedPoint;
+
+  /**
+   * Get the width and height of the Token in pixels.
+   * @param data - The width and/or height in grid units (must be positive)
+   * @returns The width and height in pixels
+   */
+  getSize(data?: Partial<{ width: number; height: number }>): { width: number; height: number };
```

Comment:
Please interface-ify.

Reply:
As all things in life, it depends. I'd say if you're going to leave comments on the interface then you need to have two interfaces. However one interface most likely suffices, without comment.